### ISSUE TITLE
jasmine.clock().tick() sets Date to destination for all interim events

### ISSUE THREAD

**Issue by ryanswood** [jasmine/jasmine#915]
Here's a simple test that demonstrates the problem.

```javascript
describe('mocking time', function(){
  beforeEach(function() {
    jasmine.clock().install();
    jasmine.clock().mockDate(
      new Date(2015, 4, 20)
    );
  });

  afterEach(function() {
    jasmine.clock().uninstall();
  });

  it('sets time correctly for interim functions', function(){
    var actualTimes = [];
    setTimeout(function() { actualTimes.push(Date().getTime()); });
    setTimeout(function() { actualTimes.push(Date().getTime()); }, 1);
    jasmine.clock().tick(1);
    expect(actualTimes).toEqual([1432105200000, 1432105200001]);
  });
});
```

The failure we see is:

```javascript
Expected [ 1432105200001, 1432105200001 ] to equal [ 1432105200000, 1432105200001 ].
```

The [culprit is obvious](https://github.com/jasmine/jasmine/blob/750a9aac6d004a960fc752ce260600b68e06f711/lib/jasmine-core/jasmine.js#L1130). Here's the code from that link:

```javascript
mockDate.tick(millis);
delayedFunctionScheduler.tick(millis);
```

I don't see any reason for `mockDate.tick()` to exist. Instead you want something like `mockDate.advanceToMillis()` that can be called somewhere around [this line](https://github.com/jasmine/jasmine/blob/4be20794827a63ca5295848c58ffc478624ee4dc/src/core/DelayedFunctionScheduler.js#L122).

**Comment by ClayShentrup** [comment]
Really hoping this gets fixed soon. I took a look at it but it requires a significant refactoring, since the clock stubbing is currently isolated from the scheduled functions runner.

**Comment by slackersoft** [comment]
I agree, we'd like to see this fixed. I've made a tracker story for it here (https://www.pivotaltracker.com/story/show/104843046), but I would also be happy to review a pull request that implements this as well.

**Comment by gregjacobs** [comment]
Awesome, thanks for this!

**Comment by benchristel** [comment]
It looks like there's still [an open Tracker story](https://www.pivotaltracker.com/story/show/104843046) for this. Should it be closed?

**Comment by timb-pt** [comment]
I'm having issues with nested `setTimeout` calls, which I figure would be a fairly common use case in Jasmine tests. How often do you release and/or when can this fix be added to a release tag?

**Comment by mramato** [comment]
I just submitted #1190, which I think may be a new issue caused by this change.

**Comment by slackersoft** [comment]
#1190 should be fixed in the 2.5.1 release
